pkg/noderesourcetopology/score_test.go

```diff
@@ -0,0 +1,294 @@
+package noderesourcetopology
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"testing"
+
+	topologyv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+	listerv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/listers/topology/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	intstr "k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/klog/v2"
+	framework "k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1"
+)
+
+// return the name of the node with the highest score
+func findMaxScoreNode(nodeToScoreMap map[string]int64) string {
+	var max int64 = 0
+	electedNode := ""
+
+	for nodeName, score := range nodeToScoreMap {
+		if max < score {
+			max = score
+			electedNode = nodeName
+		}
+	}
+	return electedNode
+}
+
+// recalcResources substruct the amount of requested resources from
+// the node and updates the NodeResourceTopology object.
+// Since we're running on a testing environment there is no actual assignment, thus we need to
+// emulate that by updating the correct amount of available resources after each request
+func recalcResources(pod *v1.Pod, node *v1.Node) {
+	for _, container := range pod.Spec.Containers {
+		for resourceName, reqQuan := range container.Resources.Requests {
+			if allocQuan, ok := node.Status.Allocatable[resourceName]; ok {
+				allocQuan.Sub(reqQuan)
+				node.Status.Allocatable[resourceName] = allocQuan
+			}
+		}
+	}
+}
+
+// updateNrt playing the role of Kubelet here.
+// It will decide to which NUMA node, the pod will assigned.
+// What the function does:
+// 1. It will look for a NUMA node that can fit.
+// (IMPORTANT NOTE: we don't really care which NUMA it will be, since wether it's NUMA 0 or NUMA 1,
+// from the scoring plugin functionality perspective it won't make any difference)
+// 2. It will fetch the fitted NUMA node object from the NodeResourceTopology
+// 3. Update the NUMA node's available resources
+func updateNrt(pod *v1.Pod, ntr *topologyv1alpha1.NodeResourceTopology) error {
+	var fittedNumaId int = -1
```

Comment:
I would suggest upper case for acronyms llike NUMA, here and for
getNumaZoneById